src/Feed/Feed.jsx

```diff
@@ -27,23 +45,88 @@ export default function Feed() {
         });
     }
 
-    fetchConsumerData().then(() =>
-      fs
-        .collection("recipe")
-        .get()
-        .then((querySnapshot) => {
+    const query = fs
+      .collection("recipe")
+      .where("allergies", "!=", selected.allergy);
+
+    // Clear recipes as firebase will filter recipes
+    setRecipes([]);
+
+    if (selected.mealtype === "") {
+      fetchConsumerData().then(() =>
+        query.get().then((querySnapshot) => {
           querySnapshot.forEach((doc) => {
             // Filter liked/disliked recipes
             if (!recipesToFilter.includes(doc.id)) {
               setRecipes((prevState) => [...prevState, doc.data()]);
             }
           });
         })
-    );
-  }, [fs, auth.currentUser.uid]);
+      );
+    } else {
+      fetchConsumerData().then(() =>
+        query
+          .where("mealType", "==", selected.mealtype)
+          .get()
+          .then((querySnapshot) => {
+            querySnapshot.forEach((doc) => {
+              // Filter liked/disliked recipes
+              if (!recipesToFilter.includes(doc.id)) {
+                setRecipes((prevState) => [...prevState, doc.data()]);
+              }
+            });
+          })
+      );
+    }
+  }, [fs, auth.currentUser.uid, selected]);
+
+  const onPreferencesSelect = (preferenceType, selectedPref) => {
+    setSelected((prevState) => ({
+      ...prevState,
+      [preferenceType]: selectedPref,
+    }));
+  };
+
+  const Preferences = allergies.map((allergy) => (
+    <Dropdown.Item
```

Comment:
I suggest putting your dropdown in a seperate component file (because it's a component). You can simply copy and paste the form dropdown component I created and tweak it to your liking.